### PR TITLE
Create API endpoints

### PR DESCRIPTION
**Pull request by Tshobohwa**
## In this milestone I created API endpoints for performing the following requests:
- List all posts for a user.
- List all comments for a user's post.
- Add a comment to a post.

### CODE REVIEW

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @Tshobohwa @cherelemma , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has not showing your partner contributions. This milestone is a [pair-programming](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_api_endpoints.md#work-and-submission-mode) activity, please make sure that all your partners contibute to this project 👍 


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by rloterh — COMMENTED** [review]
Hi Salomon, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is not contributed by all (2) assigned pair group members. Kindly ensure that all assigned members to this project contribute evenly to this task. In the event where the other partner is not able to contribute, please report the same to the Student Success team.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Kamba56** [comment]
Hi @Tshobohwa,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that **doesn't have contributions from your coding partner.** The commits I see here are not from your coding partner. Kindly contact your coding partner and work with them or contact student success.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Tshobohwa** [comment]
> 

Hi @Kamba56 , my coding partner has contributed on this PR on the commit [[59b9731](https://github.com/Tshobohwa/webwisesage/pull/10/commits/59b973169b469b5cf869777d128405234c2ab991)](https://github.com/Tshobohwa/webwisesage/pull/10/commits/59b973169b469b5cf869777d128405234c2ab991) and the commit [[5ca3341](https://github.com/Tshobohwa/webwisesage/pull/10/commits/5ca3341cff3a4f7d296b1b686e611a4e4810a660)](https://github.com/Tshobohwa/webwisesage/pull/10/commits/5ca3341cff3a4f7d296b1b686e611a4e4810a660)

File: config/routes.rb [hunk 1]
```diff
@@ -12,6 +12,9 @@
   get "users/:author_id/posts/:id/comments/new", to: "comments#new"
   post "users/:author_id/posts/:id/comments", to: "comments#create"
   post "users/:author_id/posts/:id", to: "likes#create", as: "like_post"
+  get "api/users/:author_id/posts", to: "api/posts#index"
+  get "api/users/:author_id/posts/:id", to: "api/comments#index"
+  post "api/users/:author_id/posts/:id/comments/new", to: "api/comments#create"
```
Comment (Lizdev-05) [inline config/routes.rb]: - [ ] Please make sure that your API endpoint that adds' comment to a post - **POST :** `/users/:user_id/posts/:id/comments` :pray: 
- [ ] I suggest that your app has API endpoint that returns all comments for a post - **GET :** `/users/:user_id/posts/:id/comments`. If I get this `/users/:user_id/posts/:id/comments` route I get this error :point_down: 

image

**Review by Lizdev-05 — CHANGES_REQUESTED** [review]
Hi @Tshobohwa and @cherelemma , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
:heavy_check_mark: Professional Readme file :+1: 
:heavy_check_mark: Good commit history :+1:
:heavy_check_mark: No linter error :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Owusu-Desmond — APPROVED** [review]
Hi @Tshobohwa @codecaiine , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**